feat(swarm): allow behaviours to share addresses of peers (#4371)
paritytech/rust-libp2p: protocols/request-response/CHANGELOG.md

> ## 0.26.2 - unreleased

Reviewer: ```suggestion
## 0.26.2
```

> - Deprecate `libp2p_request_response::add_address`. `Behaviour` `addresses` field uses `libp2p_swarm::PeerAddresses`.

Reviewer: ```suggestion
- Deprecate `Behaviour::add_address` in favor of `Swarm::add_peer_address`.
```

>   See [PR 4371](https://github.com/libp2p/rust-libp2p/pull/4371)

Reviewer: ```suggestion
  See [PR 4371](https://github.com/libp2p/rust-libp2p/pull/4371).
```
